operator/gitops/argocd/pipeline-service/openshift-pipelines/tekton-config.yaml

```diff
@@ -11,10 +11,6 @@ spec:
   params:
     - name: createRbacResource
       value: "false"
-  platforms:
-    openshift:
-      pipelinesAsCode:
-        enable: false
   pipeline:
     default-service-account: appstudio-pipeline
     enable-api-fields: beta
```

Comment:
fyi I've got an on hold PR in infra deployments that is vetting `enable-api-fields: beta` in conjunction with the downstream PaC images (where I reproduced the v1beta1 / v1 hoopla with just bringing in the downstream PaC images).

Given my current understanding of the various bits, I would like to see that pass before adding on to it and moving foward with this.

Reply:
https://github.com/redhat-appstudio/infra-deployments/pull/1847

Reply:
I agree. This PR should not be merged until we've verified the update to downstream images in staging.

Reply:
So what I'm seeing in https://github.com/redhat-appstudio/infra-deployments/pull/1847 @Roming22 is that using 

v1.10.2-1

of the downstream PaC image and *NOT* using `enable-api-fields: beta` , we get the validating admission webhook error in the PaC controller I believe was discovered with your prior attempt to use the downstream image:

`{"severity":"ERROR","timestamp":"2023-05-24T17:58:33.259444929Z","logger":"pipelinesascode","caller":"events/emit.go:45","message":"There was an error starting the PipelineRun test-mvp-component-mhdf-on-pull-request-, creating pipelinerun test-mvp-component-mhdf-on-pull-request- in namespace mvp-dev-jvwe-tenant has failed.\n\nTekton Controller has reported this error: ```admission webhook \"validation.webhook.pipeline.tekton.dev\" denied the request: validation failed: must not set the field(s): spec.pipelineSpec.finally[0].taskref.params, spec.pipelineSpec.finally[0].taskref.resolver, spec.pipelineSpec.finally[1].taskref.params, spec.pipelineSpec.finally[1].taskref.resolver, spec.pipelineSpec.tasks[0].taskref.params, spec.pipelineSpec.tasks[0].taskref.resolver, spec.pipelineSpec.tasks[10].taskref.params, spec.pipelineSpec.tasks[10].taskref.resolver, spec.pipelineSpec.tasks[11].taskref.params, spec.pipelineSpec.tasks[11].taskref.resolver, spec.pipelineSpec.tasks[1].taskref.params, spec.pipelineSpec.tasks[1].taskref.resolver, spec.pipelineSpec.tasks[2].taskref.params, spec.pipelineSpec.tasks[2].taskref.resolver, spec.pipelineSpec.tasks[3].taskref.params, spec.pipelineSpec.tasks[3].taskref.resolver, spec.pipelineSpec.tasks[4].taskref.params, spec.pipelineSpec.tasks[4].taskref.resolver, spec.pipelineSpec.tasks[5].taskref.params, spec.pipelineSpec.tasks[5].taskref.resolver, spec.pipelineSpec.tasks[6].taskref.params, spec.pipelineSpec.tasks[6].taskref.resolver, spec.pipelineSpec.tasks[7].taskref.params, spec.pipelineSpec.tasks[7].taskref.resolver, spec.pipelineSpec.tasks[8].taskref.params, spec.pipelineSpec.tasks[8].taskref.resolver, spec.pipelineSpec.tasks[9].taskref.params, spec.pipelineSpec.tasks[9].taskref.resolver``` ","provider":"github","event-id":"984fb850-fa5c-11ed-886b-9a1619c704d9","event-sha":"7bbe00568feb097da6234e8cc04fb976a4edf512","event-type":"pull_request","stacktrace":"github.com/openshift-pipelines/pipelines-as-code/pkg/events.(*EventEmitter).EmitMessage\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/events/emit.go:45\ngithub.com/openshift-pipelines/pipelines-as-code/pkg/pipelineascode.(*PacRun).Run.func1\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/pipelineascode/pipelineascode.go:80"}`

Is my recollection correct @Roming22 ?

Also, I know there was some questions and acknowledgements that the explanations of what `enable-api-fields` does exactly were a little unclear.  Were all those sorted out?

So then, after running with `enable-api-fields: beta` set in the tektonconfig, that webhook validation error disappears from the PaC controller, but now we consistently see errors across the RHTAP e2e's in the `tekton-pipelines-controller` , where it can no longer retrieve tasks from taskbundles that it was able to retrieve before that `enable-api-fields` setting landed:

`{"severity":"error","timestamp":"2023-05-24T19:37:01.167Z","logger":"tekton-pipelines-controller","caller":"pipelinerun/pipelinerun.go:250","message":"Reconcile error: Couldn't retrieve Task \"\": failed to get task: error requesting remote resource: error getting \"bundleresolver\" \"e2e-demos-emzo-tenant/bundles-addade0469959bfe0c2e76dccc5baedd\": could not find object in image with kind: Task and name: prefetch-dependencies","commit":"048bf9a","knative.dev/controller":"github.com.tektoncd.pipeline.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"6a7dd040-9826-4de3-b847-5e6cf1d67333","knative.dev/key":"e2e-demos-emzo-tenant/devfile-sample-python-basic-avup-9jx66","stacktrace":"github.com/tektoncd/pipeline/pkg/reconciler/pipelinerun.(*Reconciler).ReconcileKind\n\t/go/src/github.com/tektoncd/pipeline/pkg/reconciler/pipelinerun/pipelinerun.go:250\ngithub.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1beta1/pipelinerun.(*reconcilerImpl).Reconcile\n\t/go/src/github.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1beta1/pipelinerun/reconciler.go:236\nknative.dev/pkg/controller.(*Impl).processNextWorkItem\n\t/go/src/github.com/tektoncd/pipeline/vendor/knative.dev/pkg/controller/controller.go:542\nknative.dev/pkg/controller.(*Impl).RunContext.func3\n\t/go/src/github.com/tektoncd/pipeline/vendor/knative.dev/pkg/controller/controller.go:491"} `

Now, should that`enable-api-fields` setting have any effect on resolving tasks from bundles?  No idea.  IMO we need openshift-pipelines team assitance.

@vdemeester @chmouel - tagging you directly here in case you can infer enough context and can provide guidance.  Thanks.

Otherwise @Roming22 - I suggest you ping #forum-tektoncd-pipelines at the first opportunity and get the discussion going.

I suppose there is also the possibility that having the openshift-pipelines operator handle all the installation (vs. just switching from the upstream to downstream image, but continuing to apply chains and PaC) addresses whatever is wrong, and both errors go away.

Reply:
QE tells me the couldnt get task is unrelated to our changes here:  https://issues.redhat.com/browse/SRVKP-2749

assuming we are all good with everything else wrt `enable-api-fields` I'll retest there

Reply:
turns out `enable-api-fields` does have an impact on bundle resolution

see https://redhat-internal.slack.com/archives/C032EJ007C0/p1685038679501169 for all the triage